cpp/tests/c_api/c_test_utils.h

```diff
@@ -20,10 +20,12 @@
 #include <stdio.h>
 #include <time.h>
 
-#define TEST_ASSERT(RETURN_VALUE, STATEMENT, MESSAGE)                    \
-  {                                                                      \
-    (RETURN_VALUE) = !(STATEMENT);                                       \
-    if ((RETURN_VALUE)) { printf("ASSERTION FAILED: %s\n", (MESSAGE)); } \
+#define TEST_ASSERT(RETURN_VALUE, STATEMENT, MESSAGE)                      \
+  {                                                                        \
+    if (!(RETURN_VALUE)) {                                                 \
+      (RETURN_VALUE) = !(STATEMENT);                                       \
+      if ((RETURN_VALUE)) { printf("ASSERTION FAILED: %s\n", (MESSAGE)); } \
+    }                                                                      \
```

Comment:
This is better than before since it won't clobber a prior failing return value, but maybe it's worthwhile to see all failing tests instead of just the first one.  Would this work for that?
```suggestion
   if (!(STATEMENT)) {  \
      printf("ASSERTION FAILED: %s\n", (MESSAGE));  \
      (RETURN_VALUE) |= 1;  \
   }  \
```

Reply:
Maybe.  Isn't the gtest convention to bail out of a particular test at the first failure and move on to the next test?  For a larger test data set you could get thousands of errors from a particular test which is probably not desired.

Reply:
gtest supports either behavior, based on if you use the `EXPECT_*` or `ASSERT_*` assertions, where EXPECT will fail the test but continue running it to see everything (the behavior I described), and ASSERT will cause it to fail and return immediately.  I wasn't thinking we'd have _that many_ failures typically, but I don't feel strongly either way. Since this is named `TEST_ASSERT`, I guess the current behavior is indeed closer to gtest's `ASSERT_*`.

If you wanted to follow the gtest convention further, you could also have two different versions of the macros with the above names.  Just another idea.  My main concern was addressed (failures not reflected in the binary's exit code), so feel free to ignore this suggestion if you'd rather keep the current behavior.